### PR TITLE
Inserted all courses to Courses DB #15

### PR DESCRIPTION
**Pull request by matanpugach**

### CODE REVIEW

File: src/main/java/course/info/DataToDBLoader.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import org.json.JSONObject;
+
+import FuncTest.TestFunc.globals;
+import Utils.JsonReader;
+
+public class DataToDBLoader {
+	
+	public static void loadToDBFromJson(String jsonName) {
+		JSONArray courses = JsonReader.ReadJsonFromFile(jsonName);
+		 for (int c = 0; c < courses.length(); ++c) {
```
Comment (Itamaror) [inline src/main/java/course/info/DataToDBLoader.java]: Please do not insert big amounts of data into the database using multiple queries as this has a chance to temporarily block our azure account ,instead use batching as explained here: https://viralpatel.net/blogs/batch-insert-in-java-jdbc/

Comment (matanpugach) [reply src/main/java/course/info/DataToDBLoader.java]: @Itamaror great comment, Thanks! wil update

**Comment by matanpugach** [comment]
@yoavz1997 i didn't understand your comment.

**Comment by matanpugach** [comment]
@yoavz1997 @Itamaror Implemented both your remarks. Thanks! Also, i added the code & output of getting the descriptions.

File: src/main/java/Utils/JsonReader.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+public class JsonReader {
+
+	private static String readAll(Reader rd) throws IOException {
+//    String s = IOUtils.toString(rd), $ = s.substring(s.indexOf('\n') + 1);
+//    return $.substring(0, $.lastIndexOf('\n'));
+		return IOUtils.toString(rd);
+	}
+
+//Had to used suppress warning since the address is always available and finally close has to happen
+	@SuppressWarnings("resource")
```
Comment (yoavz1997) [inline src/main/java/Utils/JsonReader.java]: Please resolve the warning rather then suppressing it

File: src/main/java/Utils/JsonReader.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	public static JSONArray readJsonFromUrl(String url) {
+
+		InputStream $ = null;
+		try {
```
Comment (yoavz1997) [inline src/main/java/Utils/JsonReader.java]: Use try with resource https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

File: src/main/java/course/info/WebInfoExtractor.java [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+	 */
+	private static String ExctractName(List<Element> ¢, String c) {
+		return (¢.get(0).text().replaceAll("Subject Sylbus: ", "").split(" - " + c)[0]);
+
+	}
+
+	public static void writeCsvFile(String fileName, List<CourseObject> cl) {
+		FileWriter fileWriter = null;
+
+		try {
```
Comment (yoavz1997) [inline src/main/java/course/info/WebInfoExtractor.java]: Again, please use try with resource

File: src/main/java/course/info/WebInfoExtractor.java [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+				fileWriter.close();
+			} catch (IOException ¢) {
+				System.out.println("Error while flushing/closing fileWriter !!!");
+				¢.printStackTrace();
+			}
+
+		}
+	}
+
+	public static void main(String[] args) {
```
Comment (yoavz1997) [inline src/main/java/course/info/WebInfoExtractor.java]: Please comment out the main

Comment (matanpugach) [reply src/main/java/course/info/WebInfoExtractor.java]: Done.

File: src/main/java/Utils/JsonReader.java (hunk 3 above)
Comment (matanpugach) [reply src/main/java/Utils/JsonReader.java]: AWESOME! thanks!

File: src/main/java/Utils/JsonReader.java (hunk 2 above)
Comment (matanpugach) [reply src/main/java/Utils/JsonReader.java]: done 👍

File: src/main/java/course/info/WebInfoExtractor.java (hunk 4 above)
Comment (matanpugach) [reply src/main/java/course/info/WebInfoExtractor.java]: Done!

File: src/main/java/Utils/JsonReader.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+	public static JSONArray readJsonFromUrl(String url) {
+		try (InputStream $ = new URL(url).openStream()){
+			
+			try {
+				return new JSONArray(readAll(new BufferedReader(extracted($))));
+			} catch (JSONException | IOException ¢) {
+				// TODO Auto-generated catch block
+				¢.printStackTrace();
+			} finally {
```
[the remark this answers is not in the document]
Comment (matanpugach) [reply src/main/java/Utils/JsonReader.java]: You're right! god bless try-with-resourse.

**Pull request merged**